### DISCUSSION TITLE
what's the essential difference between falkordb and redisgraph?

### DISCUSSION THREAD

**Discussion by DeoLeung** [FalkorDB/FalkorDB#14] [General]
will falkordb be more like a general db that data is persisted on disk by default?

**Comment by gkorland** [comment]
FalkorDB is going to take the RedisGraph technology, extend it and relaunch it under the FalkorDB brand.
It's still going to target cases which requires low latency and as such is going to be in memory.
But, similar to RedisGraph it has the ability to store thing on disk for persistency and durability.

**Reply by davidroman0O** [reply]
I really love Redis Graph and was sad about the recent announcement, I'm glad that some people are continuing the work!
I have more questions
Will it be a product under a company?
If no, will it have possibility to sponsor?

**Reply by gkorland** [reply]
Yes, we opened a company called FalkorDB that will push this product.

**Reply by chpatel2** [reply]
So I'm assuming there are no plans to make it a hybrid architecture where it stores and queries on disk and memory. Our issue is we committed to redis graph and its end of life. Our production instance has a ton of memory allocated to it. long term this becomes unsustainable in terms of cost.

**Reply by gkorland** [reply]
We do have plans for this, we plan to "extend" our memory such that the node/edge's properties will be store on disk but the Graph topology will be kept in memory for efficiency

**Reply by chpatel2** [reply]
Thanks for replying. We are a startup and don't have a ton of resources to just redo our stack. Hoping to switch from redisgraph to falkor. Would love to chat at some point.

**Comment by flowt-au** [comment]
@gkorland:
So, I am using AOF. From the Redis docs (I couldn't find the equivalent in the Falkor docs):
```
Persistence refers to the writing of data to durable storage, such as a solid-state disk (SSD). Redis provides a range of persistence options. These include:

RDB (Redis Database): RDB persistence performs point-in-time snapshots of your dataset at specified intervals.
AOF (Append Only File): AOF persistence logs every write operation received by the server. These operations can then be replayed again at server startup, reconstructing the original dataset. Commands are logged using the same format as the Redis protocol itself.
No persistence: You can disable persistence completely. This is sometimes used when caching.
RDB + AOF: You can also combine both AOF and RDB in the same instance.
```
What new pattern are you thinking of for persistence?

Thanks,
Murray

**Reply by swilly22** [reply]
Hi Murry,
FalkorDB uses the exact same persistency methods as RedisGraph did.
So you should be able to continue on using RDB, AOF and RDB + AOF.

I'll make sure to address this in our FalkorDB docs.

**Comment by flowt-au** [comment]
@swilly22 
Yes, thanks, it is working really well. My question is what extra persistency were you thinking of introducing? Just curious.
Go well,
Murray

**Reply by flowt-au** [reply]
Oh, ok, I just saw Guy's reply in the other thread. All good.
